source/Server/Integration/JiraConnectAppClient.cs

```diff
@@ -25,29 +27,41 @@ class JiraConnectAppClient
             this.octopusHttpClientFactory = octopusHttpClientFactory;
         }
 
-        public string? GetAuthTokenFromConnectApp(ILog log)
+        public Task<string?> GetAuthTokenFromConnectApp(ILog log)
         {
             var username = installationIdProvider.GetInstallationId().ToString();
             var password = configurationStore.GetConnectAppPassword();
             return GetAuthTokenFromConnectApp(username, password?.Value, log);
```

Comment:
Our convention is to await the task here, rather than returning it to the caller, to make call stacks in error scenarios easier to work with

Reply:
Fixed